src/JSONFormatter.hpp

```diff
@@ -0,0 +1,21 @@
+# pragma once
+
+#include "IFormatter.hpp"
+
+namespace signal_estimator {
+
+class JSONFormatter : public IFormatter {
+    public:
+        void report_latency(double sw_hw, double hw, int sma_window, double hw_avg) override;
+
+        void report_losses(double loss_rate, int sma_window, double avg_loss_rate, double loss_ratio) override;
+
+        JSONFormatter();
+        
+        ~JSONFormatter();
```

Comment:
It also makes sense to mark copy ctor and assignment operator as deleted. Same for TextFormatter.